kafka-connect-transforms/src/main/java/io/tabular/iceberg/connect/transforms/JsonToMapUtils.java

```diff
@@ -0,0 +1,281 @@
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package io.tabular.iceberg.connect.transforms;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.BigIntegerNode;
+import com.fasterxml.jackson.databind.node.BinaryNode;
+import com.fasterxml.jackson.databind.node.BooleanNode;
+import com.fasterxml.jackson.databind.node.DecimalNode;
+import com.fasterxml.jackson.databind.node.DoubleNode;
+import com.fasterxml.jackson.databind.node.FloatNode;
+import com.fasterxml.jackson.databind.node.IntNode;
+import com.fasterxml.jackson.databind.node.LongNode;
+import com.fasterxml.jackson.databind.node.MissingNode;
+import com.fasterxml.jackson.databind.node.NullNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.fasterxml.jackson.databind.node.TextNode;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableMap;
+import org.apache.iceberg.relocated.com.google.common.collect.Lists;
+import org.apache.iceberg.relocated.com.google.common.collect.Maps;
+import org.apache.kafka.connect.data.Schema;
+import org.apache.kafka.connect.data.SchemaBuilder;
+import org.apache.kafka.connect.data.Struct;
+
+public class JsonToMapUtils {
+
+  private JsonToMapUtils() {
+    throw new UnsupportedOperationException("This is a utility class and cannot be instantiated");
+  }
+
+  private static final Map<Class<? extends JsonNode>, Schema> JSON_NODE_TO_SCHEMA =
+      getJsonNodeToSchema();
+
+  private static Map<Class<? extends JsonNode>, Schema> getJsonNodeToSchema() {
+    final Map<Class<? extends JsonNode>, Schema> map = Maps.newHashMap();
+    map.put(BinaryNode.class, Schema.OPTIONAL_BYTES_SCHEMA);
+    map.put(BooleanNode.class, Schema.OPTIONAL_BOOLEAN_SCHEMA);
+    map.put(TextNode.class, Schema.OPTIONAL_STRING_SCHEMA);
+    map.put(IntNode.class, Schema.OPTIONAL_INT32_SCHEMA);
+    map.put(LongNode.class, Schema.OPTIONAL_INT64_SCHEMA);
+    map.put(FloatNode.class, Schema.OPTIONAL_FLOAT32_SCHEMA);
+    map.put(DoubleNode.class, Schema.OPTIONAL_FLOAT64_SCHEMA);
+    map.put(ArrayNode.class, Schema.OPTIONAL_STRING_SCHEMA);
+    map.put(
+        ObjectNode.class,
+        SchemaBuilder.map(Schema.STRING_SCHEMA, Schema.STRING_SCHEMA).optional().build());
+    map.put(BigIntegerNode.class, Schema.OPTIONAL_STRING_SCHEMA);
```

Comment:
Looking deeper into Kafka Connect -> 

https://github.com/apache/kafka/blob/trunk/connect/api/src/test/java/org/apache/kafka/connect/data/ValuesTest.java#L826

should be bigdecimal w/ zero scale.